google-cloud-firestore/lib/google/cloud/firestore/rate_limiter.rb

```diff
@@ -0,0 +1,73 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+module Google
+  module Cloud
+    module Firestore
+      ##
+      # @private
+      class RateLimiter
+        DEFAULT_STARTING_MAXIMUM_OPS_PER_SECOND = 500.0
+        DEFAULT_PHASE_LENGTH = 300.0
+
+        attr_reader :bandwidth
+
+        ##
+        # Initialize the object
+        def initialize starting_ops: nil, phase_length: nil
+          @start_time = time
+          @last_fetched = time
+          @bandwidth = (starting_ops || DEFAULT_STARTING_MAXIMUM_OPS_PER_SECOND).to_f
+          @phase_length = phase_length || DEFAULT_PHASE_LENGTH
+        end
+
+        ##
+        # Wait till the number of tokens is available
+        # Assumes that the bandwidth is distributed evenly across the entire second.
+        #
+        # Example - If the limit is 500 qps, then it has been further broken down to 2e+6 nsec
+        # per query
+        #
+        # @return [nil]
+        def get_tokens size
```

Comment:
this method is not really getting any tokens .. may be rename it to indicate that it waits for bandwidth ?

Reply:
How about renaming it to `wait_for_tokens`?